Update website for gem5 v24.1

erin-le opened the pull request:
This PR updates the documentation on the gem5 website for gem5 version 24.1, changes commands/instructions to use the ALL build where possible, and adds instructions for running gem5 with a pre-built binary in a few places.

The documentation updates to the gem5 website are as follows:

- Updating `_pages/documentation/gem5-stdlib` parts 0-4. I tested the scripts in the tutorials and made changes if they didn't work with gem5 v24.1.
- Updating Learning gem5 part 1, parts 1, 2, 3, and 5. The slides for 1-Introduction/02-getting-started and 2-Using-gem5/01-stdlib in the 2024 gem5 bootcamp were used to update part1/part2 and part1/part3, respectively. The materials were modified to use fewer bullet points/more prose, and the description of the X86DemoBoard in 1-Introduction/02-getting-started was updated for v24.1.

erin-le committed:
website: Updating gem5-stdlib tutorials 0-4 for gem5 v24.1

This commit updates the gem5 stdlib tutorials parts 1 through 4
+ the overview page to work with gem5 version 24.1.

erin-le committed:
website: change some files to use build/ALL in examples; markdown linting

erin-le committed:
website: update part 1 of Learning gem5 based on gem5 bootcamp material

This commit updates part 1 of Learning gem5 based on
01-Introduction/02-getting-started and 02-Using-gem5/01-stdlib of
the 2024 gem5 bootcamp.

erin-le committed:
website: Remove references to builds other than ALL in gem5 website

powerjg commented:
@Harshil2107 can you review this?

Harshil2107 commented:
Yes, I will review this today.

Harshil2107 commented on _pages/documentation/gem5-stdlib/1-tutorial-hello-world.md:
````diff
@@ -95,11 +98,11 @@ Of course, in order to run a meaningful simulation, we must specify a workload f
 ```
 
-The `Resource` class takes a string which specifies which resource, from [gem5-resources](/documentation/general_docs/gem5_resources), is to be obtained for the simulation.
+The `obtain_resource` function takes a string which specifies which resource, from [gem5-resources](/documentation/general_docs/gem5_resources), is to be obtained for the simulation.
````
I think it would be nice to also mention that a specific version of the resource can also be obtained, if the `resource_version` is used along with the id.

powerjg commented:
@erin-le what is the status of this PR?

powerjg reviewed (dismissed):
I trust you :)

erin-le committed:
website: edits to gem5-stdlib documentation

This commit modifies the gem5-stdlib/2-tutorial-x86-fs
documentation to improve clarity.
